uast/transformer/semantic.go

```diff
@@ -161,96 +164,102 @@ func (c *commentElems) isTab(r rune) bool {
 	return false
 }
 
-func (c *commentElems) Split(text string) bool {
-	if c.DoTrim {
-		text = strings.TrimLeftFunc(text, unicode.IsSpace)
+func max(x, y int) int {
+    if x > y {
+        return x
+    }
+    return y
+}
+
+// Returns the first index i of runes satisfying f(runes[i]), len(runes) otherwise
+func firstIndexFunc(runes []rune, f func(r rune) bool) int {
+	for i, r := range runes {
+		if (f(r)) {
+			return i;
+		}
 	}
+	return len(runes);
```

Comment:
maybe `return -1`, how do you know if it was found? you will have to compare to `len`

Reply:
Btw. you can try to reuse a func. from std. library:
https://golang.org/src/strings/strings.go?s=18931:18982#L758

Reply:
Well, `lastIndexFunc` returns `len(runes)` for 2 reasons:
1. That way I avoid `if-elses` to check slice the string (at least in the use cases for this file)
2. It seems more consistent to me: `firstIndexFunc` loops from left to right and returns either the first index verifying the predicate or the first index out of bounds (`len(runes)`). `lastIndexFunc` does the same but from right to left, so out of bounds would be `-1`. Again this makes possible to avoid a lot of `if-elses` in the code.

The `std` version was used before, but we agreed on making the whole code work with `[] rune` instead of strings (the `std` version takes strings as argument). It can still be used of course, but it implies more conversions `[] rune <-> string`

Reply:
It's ok to use runes, my point was to take a look into std. lib, because for instance returning len(s) if you don't find a rune is kind of not intuitive (in most cases you expect -1). On the other hand I know returning a len can be handy, because you can later call
`s[:index()]`, but again it's a special case, so maybe we should just return a substring/slice instead of index.

Reply:
:thinking: It's very likely I am not having into account Go conventions. So as you wish, I can change it to `-1`s